server/src/test/java/org/elasticsearch/http/DefaultRestChannelTests.java

```diff
@@ -722,27 +788,13 @@ public HttpResponse createResponse(RestStatus status, ChunkedRestResponseBody co
                 Level.TRACE,
                 "[" + request.getRequestId() + "] response body",
                 ReferenceDocs.HTTP_TRACER,
-                () -> channel.sendResponse(RestResponse.chunked(RestStatus.OK, new ChunkedRestResponseBody() {
-
-                    boolean isDone;
-
-                    @Override
-                    public boolean isDone() {
-                        return isDone;
-                    }
-
-                    @Override
-                    public ReleasableBytesReference encodeChunk(int sizeHint, Recycler<BytesRef> recycler) {
-                        assertFalse(isDone);
-                        isDone = true;
-                        return ReleasableBytesReference.wrap(responseBody);
-                    }
-
-                    @Override
-                    public String getResponseContentTypeString() {
-                        return RestResponse.TEXT_CONTENT_TYPE;
-                    }
-                }, () -> assertTrue(isClosed.compareAndSet(false, true))))
+                () -> channel.sendResponse(
+                    RestResponse.chunked(
+                        RestStatus.OK,
+                        new TestBody(responseBody, between(0, 3)),
+                        () -> assertTrue(isClosed.compareAndSet(false, true))
```

Comment:
Could we also assert `isDone() && testBody.isEndOfReponse()` (on the test body)?

Reply:
Think so, yes, see 0d923f42bbe.

Reply:
Wait, no, or at least not very easily. We create a sequence of distinct `TestBody` instances, one for each part. I suppose we could capture them on creation but it does rather obscure the point of this test to do so. I'd rather leave these out - I think this is adequately covered elsewhere.

Reply:
Actually maybe not too bad, see a2c385ea17f.